broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java

```diff
@@ -126,73 +116,82 @@ public CompletableFuture<Void> start() {
-                      snapshotStoreFactory,
-                      clusterServices,
-                      exporterRepository,
-                      healthCheckService,
-                      diskSpaceUsageMonitor,
-                      gatewayBrokerTransport,
-                      jobStreamer);
-
-              partitions.addAll(
-                  partitionFactory.constructPartitions(
-                      partitionGroup,
-                      partitionListeners,
-                      topologyManager,
-                      brokerCfg.getExperimental().getFeatures().toFeatureFlags()));
-
-              final var futures =
-                  partitions.stream()
-                      .map(partition -> CompletableFuture.runAsync(() -> startPartition(partition)))
-                      .toArray(CompletableFuture[]::new);
-
-              CompletableFuture.allOf(futures).join();
-              return null;
-            });
+    actorSchedulingService.submitActor(topologyManager);
+    healthCheckService.registerPartitionManager(this);
+
+    final var featureFlags = brokerCfg.getExperimental().getFeatures().toFeatureFlags();
+    final var partitionFactory =
+        new PartitionFactory(
+            actorSchedulingService,
+            brokerCfg,
+            localBroker,
+            commandApiService,
+            snapshotStoreFactory,
+            clusterServices,
+            exporterRepository,
+            healthCheckService,
+            diskSpaceUsageMonitor,
+            gatewayBrokerTransport,
+            jobStreamer);
+    partitionGroup
+        .join(
+            new DefaultPartitionManagementService(
+                clusterServices.getMembershipService(), clusterServices.getCommunicationService()))
+        .forEach(
+            partitionStart ->
+                partitionStart
+                    .thenApply(
+                        partition -> {
+                          final var zeebePartition =
+                              partitionFactory.constructPartition(
+                                  partition, partitionListeners, topologyManager, featureFlags);
+                          startPartition(zeebePartition);
+                          return null;
+                        })
+                    .exceptionally(
+                        error -> {
+                          LOGGER.error("Failed to start partition", error);
+                          return null;
+                        }));
+
+    return CompletableFuture.completedFuture(null);
   }
 
   private void startPartition(final ZeebePartition zeebePartition) {
-
-    actorSchedulingService.submitActor(zeebePartition).join();
-    zeebePartition.addFailureListener(
-        new PartitionHealthBroadcaster(zeebePartition.getPartitionId(), this::onHealthChanged));
-    diskSpaceUsageMonitor.addDiskUsageListener(zeebePartition);
+    actorSchedulingService
+        .submitActor(zeebePartition)
+        .onComplete(
```

Comment:
:thought_balloon: Now we are getting a warning every time :roll_eyes: 

`No executor provided for ActorFuture#onComplete callback`